src/main/java/org/andersen/hotel/service/ApartmentService.java

```diff
@@ -0,0 +1,26 @@
+package org.andersen.hotel.service;
+
+import org.andersen.hotel.model.Apartment;
+import org.springframework.data.domain.Sort;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface ApartmentService {
```

Comment:
in general, you can create bease service with generic type

Reply:
in that case, what do i do with specific methods such as reserve apartment? have 2 interfaces (one generic and one specific) or just 1 generic and write specific methods in impl class?

Reply:
got it, delegate interface, didnt know we can extend them))

Reply:
1 base) after that new one with additional methods
and just extend it))